### PR TITLE
Moving System.Data code which needs to be consumed by Common and SqlClient to Common

### PR DESCRIPTION
**Pull request by saurabh500**
The following changes are included. 

1. Moved the FieldNameLookup and Locale related functionality to the Common/src/System/Data 
2. Modified the System.Data.SqlClient.csproj to reference the correct files.
3. Changed the FieldNameLookup to use Dictionary instead of Hashtable.

The moved classes need to be consumed by System.Data.Common and System.Data.SqlClient. 

This PR will support the changes which move DbEnumerator to the System.Data.Common and exposes it in the Contracts.

### CODE REVIEW

**Comment by saurabh500** [comment]
cc @corivera @stephentoub

File: src/Common/src/System/Data/Common/FieldNameLookup.cs [hunk 1]
```diff
@@ -92,13 +93,7 @@ public int IndexOf(string fieldName)
...
             }
             int index;
-            object value = _fieldNameLookup[fieldName];
-            if (null != value)
-            {
-                // via case sensitive search, first match with lowest ordinal matches
-                index = (int)value;
-            }
-            else
+            if(!_fieldNameLookup.TryGetValue(fieldName, out index))
```
Comment (justinvp) [inline src/Common/src/System/Data/Common/FieldNameLookup.cs]: Nit: there should be a space between `if` and the `(`

File: src/Common/src/System/Data/Common/FieldNameLookup.cs [hunk 2]
```diff
@@ -6,16 +6,16 @@
 
 //------------------------------------------------------------------------------
 
-using System.Collections;
 using System.Data.Common;
 using System.Globalization;
+using System.Collections.Generic;
```
Comment (justinvp) [inline src/Common/src/System/Data/Common/FieldNameLookup.cs]: Nit: sort usings

File: src/Common/src/System/Data/Common/FieldNameLookup.cs (hunk 1 above)
Comment (justinvp) [inline src/Common/src/System/Data/Common/FieldNameLookup.cs]: Also, would it be worth preserving the comment?

```csharp
// via case sensitive search, first match with lowest ordinal matches
if (!_fieldNameLookup.TryGetValue(fieldName, out index))
```

File: src/Common/src/System/Data/Common/FieldNameLookup.cs [hunk 3]
```diff
@@ -6,16 +6,16 @@
...
 
 namespace System.Data.ProviderBase
 {
     internal sealed class FieldNameLookup
     {
         // hashtable stores the index into the _fieldNames, match via case-sensitive
-        private Hashtable _fieldNameLookup;
+        private Dictionary<string, int> _fieldNameLookup;
```
Comment (justinvp) [inline src/Common/src/System/Data/Common/FieldNameLookup.cs]: Nit: the above comment refers to "hashtable", which could be changed to "Dictionary"

File: src/Common/src/System/Data/Common/FieldNameLookup.cs [hunk 4]
```diff
@@ -142,7 +138,7 @@ private int LinearIndexOf(string fieldName, CompareOptions compareOptions)
         private void GenerateLookup()
         {
             int length = _fieldNames.Length;
-            Hashtable hash = new Hashtable(length);
+            Dictionary<string, int> hash = new Dictionary<string, int>(length);
```
Comment (justinvp) [inline src/Common/src/System/Data/Common/FieldNameLookup.cs]: Nit: There is a comment above the `GenerateLookup` method that refers to "Hashtable" which could be changed to "Dictionary".

**Pull request merged**